lib/core/error_handeling/AppExceptions.dart

```diff
@@ -0,0 +1,17 @@
+class ParsingError implements Exception {
```

Comment:
In Dart we already have ParseException class you don't need this

Reply:
Eng Mohamed said this is better.